cpp/src/arrow/array/builder_binary.cc

```diff
@@ -79,23 +80,6 @@ Status BinaryBuilder::AppendNextOffset() {
   return offsets_builder_.Append(static_cast<int32_t>(num_bytes));
 }
 
-Status BinaryBuilder::Append(const uint8_t* value, int32_t length) {
```

Comment:
I don't think it's useful to inline those if `AppendNextOffset` is not inlined.

Reply:
Good point. I'll inline AppendNextOffset then